src/reagent/impl/util.cljs

```diff
@@ -147,7 +147,9 @@
            rst)))
 
 (defn- merge-class [p1 p2]
-  (assoc p2 :class (class-names (:class p1) (:class p2))))
+  (if-let [names (class-names (:class p1) (:class p2))]
+    (assoc p2 :class names)
+    p2))
```

Comment:
What if someone does want `:class nil` to be propagated?

Reply:
Good catch! I propose the following solution: if any of class keys exist - then add `:class` key to the result.

Reply:
Btw. I came across this issue while updating Reagent from `0.8.1` to `0.9.0`, so before 0.9 `:class nil` was omitted from the result.